include/libdnf/base/transaction.hpp

```diff
@@ -64,6 +74,33 @@ class Transaction {
     // @replaces libdnf/Goal.formatAllProblemRules(const std::vector<std::vector<std::string>> & problems);
     std::string all_package_solver_problems_to_string();
 
+    /// Retrive comment describing the transaction
+    std::optional<std::string> get_comment() const noexcept;
+
+    /// Set a user-specified comment describing the transaction
+    void set_comment(const std::string & value);
+
+    /// Get command line of a program that started the transaction
+    std::optional<std::string> get_cmdline() const noexcept;
+
+    /// Set command line of a program that started the transaction
+    void set_cmdline(const std::string & value);
+
+    /// Get UID of a user that started the transaction
+    std::optional<uint32_t> get_user_id() const noexcept;
+
+    /// Set UID of a user that started the transaction
+    void set_user_id(uint32_t value);
+
+    /// Prepare, check and run rpm transaction
+    TransactionRunResult run(libdnf::rpm::TransactionCallbacks & callbacks);
```

Comment:
Please correct me if I am wrong, `comment` is optional argument, `user_id` overrides default detection (Used by daemon), and `cmdline` is more less like required argument. We want to have all transaction with some label provided by API user, microdnf, or Daemon.
What about to use run(libdnf::rpm::TransactionCallbacks & callbacks, const std::string & cmdline) + two optional arguments?

Reply:
Well, it's not entirely clear what the `cmdline` should be for dbus API (or basically any API) use-case. All you can put there is just generic `<path to dnfdaemon-server>`.

Reply:
I think we it will be nice if we will have no transaction without a description who set it up. For gui/daemon tools it will be enough to write down the name of the client, if it will be possible (Gnome Software, ...). What do you think?

Reply:
OK, makes sense. I've removed the attributes from `Transaction` class and changed them to arguments of `run()` method.